flow/connectors/elasticsearch/elasticsearch.go

```diff
@@ -0,0 +1,210 @@
+package conn_elasticsearch
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"log/slog"
+	"net/http"
+	"sync"
+	"time"
+
+	"go.temporal.io/sdk/log"
+
+	metadataStore "github.com/PeerDB-io/peer-flow/connectors/external_metadata"
+	"github.com/PeerDB-io/peer-flow/generated/protos"
+	"github.com/PeerDB-io/peer-flow/logger"
+	"github.com/PeerDB-io/peer-flow/model"
+	"github.com/PeerDB-io/peer-flow/model/qvalue"
+	"github.com/elastic/go-elasticsearch/v8"
+	"github.com/elastic/go-elasticsearch/v8/esutil"
+	"github.com/google/uuid"
+)
+
+type ElasticsearchConnector struct {
+	pgMetadata *metadataStore.PostgresMetadata
+	client     *elasticsearch.Client
+	logger     log.Logger
+}
+
+func NewElasticsearchConnector(ctx context.Context,
+	config *protos.ElasticsearchConfig,
+) (*ElasticsearchConnector, error) {
+	esCfg := &elasticsearch.Config{
+		Addresses: config.Addresses,
+		Transport: &http.Transport{
+			MaxIdleConnsPerHost: 4,
+			TLSClientConfig: &tls.Config{
+				MinVersion: tls.VersionTLS13,
+			},
+		},
+	}
+	if config.AuthType == protos.ElasticsearchAuthType_BASIC {
+		esCfg.Username = *config.Username
+		esCfg.Password = *config.Password
+	} else if config.AuthType == protos.ElasticsearchAuthType_APIKEY {
+		esCfg.APIKey = *config.ApiKey
+	}
+
+	esClient, err := elasticsearch.NewClient(*esCfg)
+	if err != nil {
+		return nil, fmt.Errorf("error creating elasticsearch connector: %w", err)
+	}
+	pgMetadata, err := metadataStore.NewPostgresMetadata(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	return &ElasticsearchConnector{
+		client:     esClient,
+		pgMetadata: pgMetadata,
+		logger:     logger.LoggerFromCtx(ctx),
+	}, nil
+}
+
+func (esc *ElasticsearchConnector) ConnectionActive(ctx context.Context) error {
+	err := esc.client.DiscoverNodes()
+	if err != nil {
+		return fmt.Errorf("failed to check if elasticsearch peer is active: %w", err)
+	}
+	return nil
+}
+
+func (esc *ElasticsearchConnector) Close() error {
+	// stateless connector
+	return nil
+}
+
+func (esc *ElasticsearchConnector) IsQRepPartitionSynced(ctx context.Context,
+	req *protos.IsQRepPartitionSyncedInput,
+) (bool, error) {
+	return esc.pgMetadata.IsQRepPartitionSynced(ctx, req)
+}
```

Comment:
```suggestion
```
Not needed after flattening pgMetadata field

Reply:
adds CDC related methods which I find confusing

Reply:
don't worry, eventually when we want cdc anyways it'll already be there